PredixMobileUIKitTests/PredixBarChartViewTests.swift

```diff
@@ -0,0 +1,369 @@
+
+    func testHandleOptionEnableLegend() {
+        let barChart = PredixBarChartView()
+        barChart.handleOption(.enableLegend)
+        XCTAssertEqual(barChart.legend.enabled, true, "Option enableLegend should be true")
+    }
+
+    func testHandleOptionDisableLegend() {
+        let barChart = PredixBarChartView()
+        barChart.handleOption(.disableLegend)
+        XCTAssertEqual(barChart.legend.enabled, false, "Option disableLegend should be false")
+    }
+
+    func testHandleOptionDisableSideLabels() {
+        let barChart = PredixBarChartView()
+        barChart.handleOption(.disableSideLabels)
+        XCTAssertEqual(barChart.rightAxis.drawLabelsEnabled, false, "Right Axis drawLabelsEnabled should be false")
+        XCTAssertEqual(barChart.leftAxis.drawLabelsEnabled, false, "Left Axis drawLabelsEnabled should be false")
+    }
+
+    func testHandleOptionEnableSideLabels() {
+        let barChart = PredixBarChartView()
+        barChart.handleOption(.enableSideLabels)
+        XCTAssertEqual(barChart.rightAxis.drawLabelsEnabled, true, "Right Axis drawLabelsEnabled should be true")
+        XCTAssertEqual(barChart.leftAxis.drawLabelsEnabled, true, "Left Axis drawLabelsEnabled should be true")
+    }
+
+    func testHandleOptionToggValues() {
+        let barChart = PredixBarChartView()
+        var months: [String]!
+        months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+        let unitsSold = [20.0, 4.0, 6.0, 3.0, 12.0, 16.0, 4.0, 18.0, 2.0, 4.0, 5.0, 4.0]
+        let unitsBought = [10.0, 2.0, 20.0, 4.0, 5, 8.0, 9.0, 15.0, 1.0, 3.0, 10.0, 18.0]
+
+        let bar1 = Bar(unitsSold, label: "Units Sold", colors: [NSUIColor.blue])
+        let bar2 = Bar(unitsBought, label: "Units Bought", colors: [NSUIColor.red])
+
+        barChart.create(xAxisValues: months, bars: [bar1, bar2], showWithDefaultAnimation: true)
+        XCTAssertTrue(barChart.data?.dataSetCount ?? 0 > 0, "No datasets were loaded in prepareForInterfaceBuilder")
+
+        barChart.handleOption(.toggleValues)
+        XCTAssertEqual(barChart.data?._dataSets[0].drawValuesEnabled, false, "Value on a the first data set should be set to false")
+
+        barChart.handleOption(.toggleValues)
+        XCTAssertEqual(barChart.data?._dataSets[0].drawValuesEnabled, true, "Value on a the first data set should be set to true")
+    }
+
+    func testHandleOptionToggleBarBorders() {
+        let barChart = PredixBarChartView()
+        barChart.create(xAxisValues: months, bars: [bar1, bar2], showWithDefaultAnimation: true)
+        XCTAssertTrue(barChart.data?.dataSetCount ?? 0 > 0, "No datasets were loaded")
+
+        barChart.handleOption(.toggleBarBorders)
+
+        for set in barChart.data!.dataSets {
+            let get = set as? BarChartDataSet
+            XCTAssertEqual(get?.barBorderWidth, 1.0, "Each bar border should be 1.0 ")
+        }
+    }
+
+    func testHandleOptionAnimateXY() {
```

Comment:
It's good these animation tests ensure the animation handler is called, but they're not confirming the parameters are valid. How do you know the .animateXY animation is not calling the animator with .animateX animation? Would be good to add some validation in these.

Reply:
Ok, now they are confirming the parameters are valid.